### PR TITLE
Updated login/register/forgot password

### PR DESCRIPTION
**Pull request by IvansZuks**

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
Please add space after the word 'benefits:' ('...benefits: check..') 
Now nothing happens if user submitted empty form => Please, return the error 'This field is required!' when submitting empty form (or empty field stays in this form). 
https://drive.google.com/file/d/1rGTRwhlpGKlcPEpcYJs__vSBWV6idsuZ/view?usp=sharing

**Comment by IvansZuks** [comment]
@aleksandrakorolova Spaces are added. Validation of form is SWPWA v4 issue and it's not connected with my code

File: packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        const { device } = this.props;
+
+        if (device.isMobile) {
+            return this.renderForgotPassword();
+        }
+
+        return (
+            <div block="ForgotPassword" elem="ContainerWrapper">
+                <h3>{ __('Forgot Your Password?') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js:63]: See final review

File: packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+/** @namespace Scandipwa/Route/ForgotPassword/Component/ForgotPasswordComponent */
+export class ForgotPasswordComponent extends MyAccountOverlay {
+    renderSignInWrapper() {
+        const { onLoginClick } = this.props;
+
+        return (
+            <div block="ForgotPassword" elem="SignInWrapper">
+                <h3>{ __('Registered Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js:30]: See final review

File: packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        );
+    }
+
+    renderCreateAccountWrapper() {
+        const { onCreateAccountClick } = this.props;
+
+        return (
+            <div block="ForgotPassword" elem="CreateAccountWrapper">
+                <h3>{ __('New Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/ForgotPassword/ForgotPassword.component.js:43]: See final review

File: packages/scandipwa/src/route/CreateAccount/CreateAccount.style.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            @include desktop {
+                max-width: 300px;
+                width: 100%;
+            }
+        }
+
+        hr {
+            border: 1px solid var(--color-botticelli);
+            margin-bottom: 20px;
+        }
```
Comment (yeegor) [inline packages/scandipwa/src/route/CreateAccount/CreateAccount.style.scss:51]: See final review

File: packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                { this.renderCreateAccount() }
+            </div>
+        );
+    }
+
+    renderForgotPasswordWrapper() {
+        return (
+            <div block="CreateAccount" elem="ForgetPasswordWrapper">
+                <h3>{ __('Forgot Your Password?') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js:55]: See final review

File: packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                <button block="Button" onClick={ onLoginClick }>{ __('Sign In Here') }</button>
+            </div>
+        );
+    }
+
+    renderCreateAccountWrapper() {
+        return (
+            <div block="CreateAccount" elem="CreateAccountWrapper">
+                <h3>{ __('New Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js:41]: See final review

File: packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+/** @namespace Scandipwa/Route/CreateAccount/Component/CreateAccountComponent */
+export class CreateAccountComponent extends MyAccountOverlay {
+    renderSignInWrapper() {
+        const { onLoginClick } = this.props;
+
+        return (
+            <div block="CreateAccount" elem="SignInWrapper">
+                <h3>{ __('Registered Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/CreateAccount/CreateAccount.component.js:30]: See final review

File: packages/scandipwa/src/route/ForgotPassword/ForgotPassword.style.scss [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+        > div {
+            flex: 1;
+            margin: 0 10px;
+        }
+
+        hr {
+            border: 1px solid var(--color-botticelli);
+            margin-bottom: 20px;
+        }
```
Comment (yeegor) [inline packages/scandipwa/src/route/ForgotPassword/ForgotPassword.style.scss:60]: See final review

File: packages/scandipwa/src/route/LoginAccount/LoginAccount.component.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        const { device } = this.props;
+
+        if (device.isMobile) {
+            return this.renderSignIn();
+        }
+
+        return (
+            <div block="LoginAccount" elem="SignInWrapper">
+                <h3>{ __('Registered Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/LoginAccount/LoginAccount.component.js:34]: See final review

File: packages/scandipwa/src/route/LoginAccount/LoginAccount.component.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                        { __('Create an Account') }
+                    </button>
+                </div>
+            );
+        }
+
+        return (
+            <div block="LoginAccount" elem="CreateAccountWrapper">
+                <h3>{ __('New Customers') }</h3>
+                <hr />
```
Comment (yeegor) [inline packages/scandipwa/src/route/LoginAccount/LoginAccount.component.js:62]: See final review

File: packages/scandipwa/src/route/LoginAccount/LoginAccount.container.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        ...this.containerFunctions,
+        onCreateAccountClick: this.onCreateAccountClick.bind(this)
+    };
+
+    onCreateAccountClick() {
+        history.replace(appendWithStoreCode(`/${ REGISTER }`));
+    }
+
+    handleForgotPassword() {
+        history.replace(appendWithStoreCode(`/${ FORGOT_PASSWORD }`));
```
Comment (yeegor) [inline packages/scandipwa/src/route/LoginAccount/LoginAccount.container.js:38]: Won't this work without `/`? Just `appendWithStoreCode(FORGOT_PASSWORD)`

File: packages/scandipwa/src/route/LoginAccount/LoginAccount.style.scss [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+        h4 {
+            padding-bottom: 7px;
+            text-align: center;
+        }
+
+        hr {
+            border: 1px solid var(--color-botticelli);
+            margin-bottom: 20px;
+        }
```
Comment (yeegor) [inline packages/scandipwa/src/route/LoginAccount/LoginAccount.style.scss:75]: See final review

**Review by yeegor — CHANGES_REQUESTED** [review]
Overall, very nice :)
Just one small thing:

1. Prefer using CSS over `hr`, these HTML elements is more of a deprecated legacy than a good practice

**Review by yeegor — APPROVED** [review]
Good!

**Pull request merged**